x/pki/tests/test-design.md

```diff
@@ -0,0 +1,276 @@
+## [Add DA Root](./handler_add_paa_cert_test.go)
+
+### Propose adding of DA root certificate
+
+Indexes to check:
+
+* Present:
+    * `ProposedCertificate`
+    * `UniqueCertificate`
+* Missing:
+    * `RejectedCertificate`
+    * `All Certificates`: Subject+SKID, SKID, Subject
+    * `DA Certificates`: Subject+SKID (approved), Subject+SKID (root), SKID, Subject
+
+Test cases:
+
+* Positive:
+    * Propose adding of DA root certificate: `TestHandler_ProposeAddDaRootCert`
+    * Propose adding of previously rejected DA root certificate: ?
+    * Propose adding of DA root certificate with same Subject/SKID as existing Approved certificate but different Serial
+      Number: `TestHandler_ProposeAddX509RootCert_ForDifferentSerialNumber` (need to rewrite)
+* Negative:
+    * TBD
+
+### Propose and approve adding of DA root certificate
+
+Indexes:
+
+* Present:
+    * `UniqueCertificate`
+    * `All Certificates`: Subject+SKID, SKID, Subject
+    * `DA Certificates`: Subject+SKID (approved), Subject+SKID (root), SKID, Subject
+* Missing:
+    * `ProposedCertificate`
+
+Test cases:
+
+* Positive:
+    * Propose add approve adding of DA root certificate: `TestHandler_AddDaRootCert`,
+      `TestHandler_AddDaRootCert_TwoThirdApprovalsNeeded`,
+      `TestHandler_AddDaRootCert_FourApprovalsAreNeeded_FiveTrustees`
+* Negative:
+    * TBD
+
+### Propose and reject adding of DA root certificate
+
+Indexes:
+
+* Present:
+    * `RejectedCertificate`
+* Missing:
+    * `ProposedCertificate`
+    * `UniqueCertificate`
+    * `All Certificates`: Subject+SKID, SKID, Subject
+    * `DA Certificates`: Subject+SKID (approved), Subject+SKID (root), SKID, Subject
+
+Test cases:
+
+* Positive:
+    * Propose add reject adding of DA root certificate: `TestHandler_RejectAddDaRootCert`,
+* Negative:
+    * TBD
+
+## [Add DA Intermediate](./handler_add_pai_cert_test.go)
+
+Indexes to check:
+
+* Present:
+    * `UniqueCertificate`
+    * `All Certificates`: Subject+SKID, SKID, Subject
+    * `DA Certificates`: Subject+SKID (approved), SKID, Subject
+    * `ChildCertificates`: for parent
+* Missing:
+    * `ProposedCertificate`
+
+Test cases:
+
+* Positive:
+    * Add DA intermediate certificate: `TestHandler_AddDaIntermediateCert`
+* Negative:
+    * TBD
+
+## [Revoke DA Root](./handler_revoke_paa_cert_test.go)
```

Comment:
Can you please analyze tests for remove/revoke by SerialNumber (at least simple cases)?